src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/crud-table/crud-form/crud-form.component.ts

```diff
@@ -32,28 +37,46 @@ export class CrudFormComponent implements OnInit {
     private dataGatewayService: DataGatewayService,
     private activatedRoute: ActivatedRoute,
     private taskWrapper: TaskWrapperService,
-    private location: Location
+    private location: Location,
+    private router: Router
   ) {}
 
   ngOnInit(): void {
     this.activatedRoute.data.subscribe((data: any) => {
       this.resource = data.resource;
       this.dataGatewayService.list(`ui-${this.resource}`).subscribe((response: any) => {
-        this.title = response.forms[0].control_schema.title;
-        this.uiSchema = response.forms[0].ui_schema;
-        this.controlSchema = response.forms[0].control_schema;
+        for (let i = 0; i < response.forms.length; i++) {
+          if (this.router.url.includes(response.actions[i].routerLink)) {
+            this.urlFormName = this.router.url.substring(this.router.url.lastIndexOf('/') + 1);
+            this.methodType = response.forms[i].method_type;
+            this.title = response.forms[i].container_schema.control_schema.title;
+            this.uiSchema = response.forms[i].container_schema.ui_schema;
+            this.controlSchema = response.forms[i].container_schema.control_schema;
+          }
+        }
```

Comment:
@Pegonzal sorry but i didn't understand this one. can you elaborate on what's the intent here?

Reply:
Sure @nizamial09 !
The crud component will return a JSON object with different values, one of them being the forms and another the actions, such as `create`, `edit`...
In this case, for `ceph_users.py` it will return among other values an array of forms, something like: `forms: [create_form, edit_form]`.
And what we do here is to loop through the different forms and find the one that matches the current `router.url` to populate the different values for that form, `title`, `uiSchema`, ...

This way we set up the form dynamically depending on the current url.

Hope that makes it a little bit more clear, let me know!